pkg/app/app.go

```diff
@@ -95,42 +92,27 @@ type App interface {
 	UpdateLib(name string, env string, spec *LibraryConfig) (*LibraryConfig, error)
 	// UpdateRegistry updates a registry.
 	UpdateRegistry(spec *RegistryConfig) error
-	// Upgrade upgrades an application to the current version.
-	Upgrade(dryRun bool) error
-
+	// Upgrade upgrades an application (app.yaml) to the current version.
+	Upgrade(bool) error
 	// VendorPath returns the root of the vendor path.
 	VendorPath() string
 }
 
 // Load loads the application configuration.
 func Load(fs afero.Fs, httpClient *http.Client, appRoot string) (App, error) {
-	log := log.WithField("action", "app.Load")
-
 	spec, err := read(fs, appRoot)
 	if os.IsNotExist(err) {
 		// During `ks init`, app.yaml will not yet exist - generate a new one.
-		return NewApp010(fs, appRoot, httpClient), nil
+		return NewBaseApp(fs, appRoot, httpClient), nil
```

Comment:
could mock / override NewBaseApp to get some test coverage on `app.Load()` - not blocking just wanted to note it for future.